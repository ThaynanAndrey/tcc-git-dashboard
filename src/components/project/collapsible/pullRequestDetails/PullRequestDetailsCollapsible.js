import React from 'react';
import Collapsible from '../../../layout/Collapsible';
import PullRequestCommitsCollapsible from './PullRequestCommitsCollapsible';
import PullRequestDescriptionCollapsible from './PullRequestDescriptionCollapsible';
import PullRequestCommentsCollapsible from './PullRequestCommentsCollapsible';

/**
 * Stateless component with the function of presenting a Collapsible to Pull Request Details.
 * 
 * @author Thaynan Nunes
 */
const PullRequestDetailsCollapsible = ({ pullRequest  }) => (
    <Collapsible>
        <PullRequestDescriptionCollapsible
            pullRequest={pullRequest}
        />

        <PullRequestCommitsCollapsible
            pullRequest={pullRequest}
        />

        <PullRequestCommentsCollapsible
            pullRequest={pullRequest}
        />
    </Collapsible>
);

export default PullRequestDetailsCollapsible;