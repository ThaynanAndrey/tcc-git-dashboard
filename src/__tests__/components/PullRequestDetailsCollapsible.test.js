import React from 'react';
import { mount, configure } from "enzyme";
import toJson from 'enzyme-to-json';

import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import PullRequestDetailsCollapsible from '../../components/project/collapsible/pullRequestDetails/PullRequestDetailsCollapsible';

describe('Test PullRequestDetailsCollapsible', () => {
  let tree;
  
  beforeEach(() => {
    const pullRequest = {};

    tree = mount(
      <PullRequestDetailsCollapsible
        pullRequest={pullRequest}
      />
    );
  });

  it('renders correctly a component to show PullRequestDetailsCollapsible with correct props', () => {
    expect(toJson(tree)).toMatchSnapshot();
    expect(tree.prop('pullRequest')).toEqual({});
  });

  
  it('the component should have class collapsible-header', () => {
    expect(tree.find(".collapsible-header").exists()).toBeTruthy();
  });

  it('the component should have classes collapsible-body, body-collapsible-project and descriptionPullRequestContent, and has correc props', () => {
    const pullRequest = {
        description: "###Apenas um teste!",
        commits: [{
            author: {name: "Teste"},
            date: "10/10/2010",
            message: "nada"
        }],
        comments: [{
            author: {name: "Teste"},
            date: "10/10/2010",
            message: "nada"
        }]
    };

    tree = mount(
      <PullRequestDetailsCollapsible
        pullRequest={pullRequest}
      />
    );

    expect(tree.find(".collapsible-body.body-collapsible-project.descriptionPullRequestContent").exists()).toBeTruthy();
    expect(tree.prop('pullRequest')).toEqual(pullRequest);
  });
})