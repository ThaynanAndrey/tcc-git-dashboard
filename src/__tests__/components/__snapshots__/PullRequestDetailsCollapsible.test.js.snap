// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PullRequestDetailsCollapsible renders correctly a component to show PullRequestDetailsCollapsible with correct props 1`] = `
<PullRequestDetailsCollapsible
  pullRequest={Object {}}
>
  <Collapsible>
    <ul
      className="collapsible"
    >
      <PullRequestDescriptionCollapsible
        pullRequest={Object {}}
      >
        <CollapsibleCard
          isActive={false}
        >
          <li
            className=""
          >
            <div
              className="collapsible-header"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "space-between",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <i
                  className="material-icons"
                >
                  dehaze
                </i>
                Descrição
              </div>
            </div>
            <div
              className="collapsible-body body-collapsible-project descriptionPullRequestContent"
              id="markdown-description"
            />
          </li>
        </CollapsibleCard>
      </PullRequestDescriptionCollapsible>
      <PullRequestCommitsCollapsible
        pullRequest={Object {}}
      >
        <CollapsibleCard
          isActive={false}
        >
          <li
            className=""
          >
            <div
              className="collapsible-header"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "space-between",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "red",
                      "borderRadius": "0.5rem",
                      "color": "white",
                      "height": "27px",
                      "marginRight": "1rem",
                      "textAlign": "center",
                      "width": "2rem",
                    }
                  }
                />
                Commits
              </div>
            </div>
          </li>
        </CollapsibleCard>
      </PullRequestCommitsCollapsible>
      <PullRequestCommentsCollapsible
        pullRequest={Object {}}
      >
        <CollapsibleCard
          isActive={false}
        >
          <li
            className=""
          >
            <div
              className="collapsible-header"
              style={
                Object {
                  "display": "flex",
                  "justifyContent": "space-between",
                }
              }
            >
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "backgroundColor": "red",
                      "borderRadius": "0.5rem",
                      "color": "white",
                      "height": "27px",
                      "marginRight": "1rem",
                      "textAlign": "center",
                      "width": "2rem",
                    }
                  }
                />
                Comentários
              </div>
            </div>
          </li>
        </CollapsibleCard>
      </PullRequestCommentsCollapsible>
    </ul>
  </Collapsible>
</PullRequestDetailsCollapsible>
`;
